Classes/Hooks/Frontend/TypoScriptLanguageHook.php

```diff
@@ -45,30 +69,31 @@ class TypoScriptLanguageHook
      */
     protected function createLanguageConditions($pageId)
     {
-        $content = '';
+        $setup = '';
         $languages = LanguageUtility::getLanguageRows($pageId);
 
-        foreach ($languages as $languageUid => $languageRec) {
-            $content .= implode(LF, $this->includeContent) . LF;
-
-            $content = str_replace(self::SYS_LANGUAGE_UID_PLACEHOLDER, $languageUid, $content);
-            $content = str_replace(self::LANGUAGE_PLACEHOLDER, $languageRec['language'], $content);
-            $content = str_replace(self::LOCALE_PLACEHOLDER, $languageRec['locale'], $content);
-            $content = str_replace(self::HREF_LANG_PLACEHOLDER, $languageRec['hreflang'], $content);
-            $content = str_replace(self::DIRECTION_PLACEHOLDER, $languageRec['direction'], $content);
+        foreach ($languages as $uid => $row) {
+            $setup .= implode(LF, $this->setupTemplate) . LF;
+            $setup = str_replace(self::SYS_LANGUAGE_UID_PLACEHOLDER, $uid, $setup);
+            $setup = str_replace(self::LANGUAGE_PLACEHOLDER, $row['language'], $setup);
+            $setup = str_replace(self::LOCALE_PLACEHOLDER, $row['locale'], $setup);
+            $setup = str_replace(self::HREF_LANG_PLACEHOLDER, $row['hreflang'], $setup);
+            $setup = str_replace(self::DIRECTION_PLACEHOLDER, $row['direction'], $setup);
         }
 
-        return $content;
+        return $setup;
     }
 
     /**
      * Add TS language conditions to setup
      *
      * @param array $params
-     * @param TemplateService $tmpl
+     * @param TemplateService $templateService
      */
-    public function addLanguageConditions(&$params, &$tmpl)
+    public function addLanguageConditions(&$params, &$templateService)
     {
-        $tmpl->config[] = $this->createLanguageConditions(method_exists($tmpl, 'getRootId') ? $tmpl->getRootId() : 0);
+        $rootPage = method_exists($templateService, 'getRootId') ? $templateService->getRootId() : $params['rootLine'][0]['uid'];
+        ExtensionManagementUtility::addTypoScriptConstants(implode(LF, $this->constants));
+        ExtensionManagementUtility::addTypoScriptSetup($this->createLanguageConditions($rootPage));
```

Comment:
problem with your solution was that it was not correctly applied to all areas like typoscript inspector etc. while you are adding directly to the configuration the scripts are adding it to the default typoscript configuration that means it can be overridden more easily, is processed correctly and shows up in constants editor etc.